arduino/main/main.ino

```diff
@@ -1,46 +1,78 @@
-int pinSwitch   = 3;
-int switchS = 0;
-int muse1 = 0;
-int motorPin1 = 10;
-int motorSpeed1 = 0;
-int muse2 = 0;
-int motorPin2 = 11;
-int motorSpeed2 = 0;
+#include <Servo.h>
+#define NB_PLAYERS 2
 
+int MIN_FAN_SPEED = 50;
+Servo s1;
+Servo s2;
 
-void setup() { 
+struct Player {
+  Servo servo;
+  int fanSpeed;
+  int fanPin;
+  int laserPin;
+  int detectorPin;
+  int servoPin;
+  int servoValue;
+};
+Player players[NB_PLAYERS] = {
+  { s1, 255, 3, 7, 12, 5, 0},
+  { s2, 255, 11, 4, 2, 6, 180}
+};
+
+void setup() {
+  initializePins();
   Serial.begin(9600);
-  Serial.print("Hello Computer");
 }
 
-void loop() { 
-  if (Serial.available()) {
-    //Serial.println("RECEIVED");
-    String data = Serial.readString();
-    String dataP1 = data.substring(0,3);
-    String dataP2 = data.substring(3,6);
-    int museDataP1 = dataP1.toInt();
-    int museDataP2 = dataP2.toInt();
-    if (museDataP1 >= 100 and museDataP2 >= 100) {
-      motorSpeed1 = museDataP1 - 50;
-      Serial.println("Muse 1");
-      Serial.println(motorSpeed1);
-      motorSpeed2 = museDataP2 - 50;
-      Serial.println("Muse 2");
-      Serial.println(motorSpeed2);
-     }
+void initializePins() {
+  for(int i = 0; i < NB_PLAYERS; i++) {
+    pinMode(players[i].fanPin, OUTPUT);
+    pinMode(players[i].laserPin, OUTPUT);
+    pinMode(players[i].detectorPin, INPUT);
+    digitalWrite(players[i].laserPin, HIGH);
+    analogWrite(players[i].fanPin, players[i].fanSpeed);
+    players[i].servo.attach(players[i].servoPin);
+    players[i].servo.write(players[i].servoValue);
+  }
+}
+
+void loop() {
+  //  Lasers are currently broken
+  //  for(int i = 0; i < NB_PLAYERS; i++) {
+  //    checkEndGame(players[i], i);
+  //  }
+
+  waitUntilDataIsAvailable();
+  updateFanSpeed();
+  delay(100);
+}
+
+void checkEndGame(Player player, int id) {
```

Comment:
id could be name index or playerNumber no?

Reply:
hmm i think it's clear enough

Reply:
i'd got rid of it since I've defined an initial servo position for each player